### PR TITLE
chore: Flatten left nav in preparation to refactor drag to reorder

### PR DESCRIPTION
**Pull request by thenanyu**
a bit of a messy commit history but I think it's the right approach.
- main concept is to flatten nested components into a non-nested html structure to play nice with `react-beautiful-dnd`
- also adds more clear visual representation of drag to import

### CODE REVIEW

File: app/components/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ function Sidebar({ location, children }: Props) {
     if (location !== previousLocation) {
       ui.hideMobileSidebar();
     }
-  }, [ui, location]);
+  }, [ui, location, previousLocation]);
```
Comment (thenanyu) [inline app/components/Sidebar/Sidebar.js:29]: I had a linter warning here, it's generally unrelated to the rest of the changes

File: app/components/Sidebar/components/CollectionLink.js [hunk 2]
```diff
@@ -34,61 +31,63 @@ class CollectionLink extends React.Component<Props> {
...
+              key={node.id}
+              node={node}
+              collection={collection}
+              activeDocument={activeDocument}
+              prefetchDocument={prefetchDocument}
+              canUpdate={canUpdate}
+              depth={1.5}
+            />
+          ))}
+      </>
```
Comment (thenanyu) [inline app/components/Sidebar/components/CollectionLink.js:90]: 🍖 nested/recursive calls to Collection and Document Link do not have nested HTML elements

File: app/components/Sidebar/components/DocumentLink.js [hunk 3]
```diff
@@ -25,139 +23,150 @@ type Props = {|
...
-const DocumentChildren = styled(Flex)``;
+const Disclosure = styled(CollapsedIcon)`
+  position: absolute;
+  left: -24px;
+
+  ${({ expanded }) => !expanded && "transform: rotate(-90deg);"};
+`;
 
-export default DocumentLink;
+const ObservedDocumentLink = observer(DocumentLink);
```
Comment (thenanyu) [inline app/components/Sidebar/components/DocumentLink.js:171]: 🍗 the recursive call needs to be wrapped with observer; this seems like the least mysterious way to do it

File: app/components/Sidebar/components/SidebarLink.js [hunk 4]
```diff
@@ -29,46 +26,25 @@ function SidebarLink({
...
-      setExpanded(!expanded);
-    },
-    [expanded]
-  );
-
-  const handleExpand = React.useCallback(() => {
-    setExpanded(true);
-  }, []);
-
-  const showDisclosure = !!children && !hideDisclosure;
```
Comment (thenanyu) [inline app/components/Sidebar/components/SidebarLink.js:71]: 🍖 all of this logic is moved into `DocumentLink`, which was the only place that was using it at all

**Review by tommoor — APPROVED** [review]
love it, this should provide the foundation we need to finally build a real draggable sidebar.

**Pull request merged**